### PR TITLE
adds a use case integration test that runs a workflow and validates SLS (useful as a reference for future contributions)

### PR DESCRIPTION
**Pull request by jacobsalmela**
# Summary and Scope

> This PR/commit can be used as a reference for adding future use-cases

- adds a shellspec file that: starts a session, imports SLS, adds a cabinet, and stops a session
- validates the x1000 now exists in SLS by way of tavern

# Risks and Mitigations
 

Low

### CODE REVIEW

**Review by shunr-hpe — APPROVED** [review]

File: spec/integration/tavern/test_add_cabinet_ex3000.tavern.yml [hunk 1]
```diff
@@ -0,0 +1,385 @@
...
+                            sequence:
+                              - type: int
+
+---
+test_name: ensure hardware exists and has cani metadata
+
+stages: 
+  - name: ensure x1000 cabinet exists and has cani metadata
+    request:
+      url: https://localhost:8443/apis/sls/v1/hardware/x1000
```
Comment (rsjostrand-hpe) [inline spec/integration/tavern/test_add_cabinet_ex3000.tavern.yml:288]: Could we have this loop over items that we expected to be created? Like verify all of the expected cabinet/chassis/chassisbmc are present for example. I'll try to dig up an example of how to do this.

Comment (rsjostrand-hpe) [reply spec/integration/tavern/test_add_cabinet_ex3000.tavern.yml:288]: Docs: https://tavern.readthedocs.io/en/latest/basics.html?highlight=parametrize#parametrize

Examples: 
- https://github.com/Cray-HPE/hms-smd/blob/ba4d0343f3978a11c9400f999492e4e7f851bb71/test/ct/api/3-disruptive/test_components_negative.tavern.yaml#L128
- More examples: https://github.com/search?q=org%3ACray-HPE+parametrize+language%3AYAML&type=code&l=YAML

Comment (jacobsalmela) [reply spec/integration/tavern/test_add_cabinet_ex3000.tavern.yml:288]: I'd like to yes.  There are some docs on looping/enumerating.  I couldn't quite grasp all the concepts yet, but I am willing to get it going

Comment (rsjostrand-hpe) [reply spec/integration/tavern/test_add_cabinet_ex3000.tavern.yml:288]: I think something like this is what we want:
```yaml
test_name: ensure hardware exists and has cani metadata
marks:
  - parametrize:
      key: xname
      vals:
      - x1000
      - x1000c0
      - x1000c0b0
      - x1000c1
      - x1000c1b0
      - x1000c2
      - x1000c2b0
      - x1000c3
      - x1000c3b0
      - x1000c4
      - x1000c4b0
      - x1000c5
      - x1000c5b0
      - x1000c6
      - x1000c6b0
      - x1000c7
      - x1000c7b0

stages: 
  - name: ensure x1000 cabinet exists and has cani metadata
    request:
      url: https://localhost:8443/apis/sls/v1/hardware/{xname}
```

**Pull request merged**